src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+    return SUCCESS;
+}
+/**
+ * @brief Remove a particular card from the given deck
+ * Can be used to discard a particular card - during each turn
+ * Works for both the computer as well as human deck of cards
+ * 
+ * @param pp_head Pointer to the head of the deck from which card has to be removed
+ * @param card_to_be_removed  Card to be removed
+ * @return ret_type_e RET_EMPTY_LIST- If the deck has no cards
+ *                 RET_FAILURE - If the removing of card fails
+ *                 RET_SUCCESS - If the card is been removed successfully
+ */
+ret_type_e remove_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_removed)
+{
+    Deck_t *to_delete = *pp_head;
+    Deck_t *prev = *pp_head;
+    if (*pp_head == NULL)
+    {
+        printf("List is already empty.");
+        return RET_EMPTY_LIST;
+    }
+    while (to_delete->next != NULL)
+    {
+        if (to_delete->card.color == card_to_be_removed->color && to_delete->card.name == card_to_be_removed->name)
+        {
+            prev->next = to_delete->next;
+            free(&(to_delete->card));
+            return SUCCESS;
+        }
+        to_delete = to_delete->next;
+    }
+    return RET_FAILURE;
+}
+
+/**
+ * @brief Function to map the human player input to Card type structure
+ * Should be called only after validation of the user input
+ * 
+ * If user input is 3R, It will be mapped as follows
+ *  card.color=Red, card.name=3
+ * 
+ * @param input Value entered by the human during his turn
+ * 
+ * @return Card_t* Contains the mapped card name and color values
+ * If Card_t is NULL it means the input is an invalid one
+ */
+Card_t *map_user_input(char *input)
+{
+
+    Card_t *temp_card = (Card_t *)malloc(sizeof(Card_t));
+    temp_card = NULL;
+
+    if (input[0] == 'R')
+    {
+        temp_card->color = RED;
+    }
+    else if (input[0] == 'G')
+    {
+        temp_card->color = GREEN;
+    }
+    else if (input[0] == 'B')
+    {
+        temp_card->color = BLUE;
+    }
+
+    else if (input[0] == 'Y')
+    {
+        temp_card->color = YELLOW;
+    }
+    else
+    {
+        return temp_card;
+    }
+    //To map the number part of the User input to the card name field
+    temp_card->name = input[1] - '0';
+    return temp_card;
+}
+/**
+ * @brief Function picks a particular card from the current human deck of cards
+ * Called when a particular card has to be selected for removing from a deck
+ * Frees the menory allocated to he temporary selected drop card input parameter
+ * Note: Function  must be called only if the card to be selected is VALID
+ * 
+ * @param pp_head :Pointer which points to the address of head of card list in the Deck
+ * @param card_to_be_matched :Card which has to be matched from the deck
+ * 
+ * @return Card_t* Card that has been matched with the given card value
+ *         NULL - Returns NULL if the given card does not match with any of the cards in the deck
+ */
+Card_t *pick_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_matched)
```

Comment:
Actually, this function does the same thing as "remove_card_from_deck", you can make a common function.

Reply:
No, this is different. Pick card just matches the human entered card with his deck - which can be removed later, ONLY IF the card is a valid one during the turn.

Remove card - removes the particular from the deck

Reply:
This has been modified now based on our discussion